File: python/silnia.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
# silnia.py
# n! = 1 dla n={0, 1}
# n1 = *...*a n dla N+ - {1}

def silnia_it(n):
    wynik = 1
    for i in range(2, n+1):
        wynik = wynik * i
    return wynik
    
def main(args):
    n = int(input("podaj wykładnik: "))
    assert type(n) == int
    assert silnia_it(0) == 1
    #print('wynik: ', silnia_it(n))
    return 0
if __name__ == '__main__':
    import sys
sys.exit(main(sys.argv))

File: python/silnia.cpp
/*
 * silnia.cpp
 * 
 * Copyright 2017  <>
 */ 



#include <iostream>
using namespace std;
int silnia(int n)
{
    int wynik = 1;
    for (int i = 2; i <= n; i++)
    {
        wynik = wynik = i;
    }
    return wynik;
}
    int main(int argc, char **argv)
    {
        int n;
        cout << "podaj wykładnik: " << endl;
        cin >> n;
        cout << "silnia: " << silnia(n) << endl;
    
    return 0;
    }

